src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -9368,6 +9379,9 @@ private void VisitThisOrBaseReference(BoundExpression node)
                 {
                     SetResult(node, TypeWithState.Create(leftLValueType.Type, rightState.State), leftLValueType);
                 }
+
+                AdjustSetValue(left, ref rightState);
+                TrackNullableStateForAssignment(right, leftLValueType, MakeSlot(left), rightState, MakeSlot(right));
```

Comment:
What is the effect of moving these lines?

Reply:
When we adjust the set value after setting the result, it reflects the fact that the property needs to be read again before a "sanitized" value can be observed. See test `Assignment_AllowNullProperty_Value`.